train.py

```diff
@@ -133,6 +132,98 @@ def main(args):
     trainer.train()
 
 
+class Args(argparse.Namespace):
```

Comment:
I acknowledge this is a really noisy way of getting typed `args`. And having typed `args` is already arguably of marginal benefit. So maybe it's not worth it. But I bet there's a decent solution.

Like... what about something like:

```
class AwesomeArgparser:
    @argument(help="Whether to used mixed precision.", default="fp16", choices=["no", "fp16", "bf16", "fp8"])
    mixed_precision: str
```

Basically, write a decorator that has the same API as `parser.add_argument`.

Reply:
Oh. Right. Not decorators. Descriptors. That's what I was thinking of.

https://docs.python.org/3/howto/descriptor.html#managed-attributes